In [79]:
import pandas as pd
import numpy as np
import plotly

In [80]:
df_match = pd.read_csv("matches.csv")

In [81]:
df_match.head(10)

id  season       city        date                        team1  \
0   1    2017  Hyderabad  2017-04-05          Sunrisers Hyderabad   
1   2    2017       Pune  2017-04-06               Mumbai Indians   
2   3    2017     Rajkot  2017-04-07                Gujarat Lions   
3   4    2017     Indore  2017-04-08       Rising Pune Supergiant   
4   5    2017  Bangalore  2017-04-08  Royal Challengers Bangalore   
5   6    2017  Hyderabad  2017-04-09                Gujarat Lions   
6   7    2017     Mumbai  2017-04-09        Kolkata Knight Riders   
7   8    2017     Indore  2017-04-10  Royal Challengers Bangalore   
8   9    2017       Pune  2017-04-11             Delhi Daredevils   
9  10    2017     Mumbai  2017-04-12          Sunrisers Hyderabad   

                         team2                  toss_winner toss_decision  \
0  Royal Challengers Bangalore  Royal Challengers Bangalore         field   
1       Rising Pune Supergiant       Rising Pune Supergiant         field   
2        Kolkata Knight Riders        Kolkata Knight Riders         field   
3              Kings XI Punjab              Kings XI Punjab         field   
4             Delhi Daredevils  Royal Challengers Bangalore           bat   
5          Sunrisers Hyderabad          Sunrisers Hyderabad         field   
6               Mumbai Indians               Mumbai Indians         field   
7              Kings XI Punjab  Royal Challengers Bangalore           bat   
8       Rising Pune Supergiant       Rising Pune Supergiant         field   
9               Mumbai Indians               Mumbai Indians         field   

   result  dl_applied                       winner  win_by_runs  \
0  normal           0          Sunrisers Hyderabad           35   
1  normal           0       Rising Pune Supergiant            0   
2  normal           0        Kolkata Knight Riders            0   
3  normal           0              Kings XI Punjab            0   
4  normal           0  Royal Challengers Bangalore           15   
5  normal           0          Sunrisers Hyderabad            0   
6  normal           0               Mumbai Indians            0   
7  normal           0              Kings XI Punjab            0   
8  normal           0             Delhi Daredevils           97   
9  normal           0               Mumbai Indians            0   

   win_by_wickets player_of_match                                      venue  \
0               0    Yuvraj Singh  Rajiv Gandhi International Stadium, Uppal   
1               7       SPD Smith    Maharashtra Cricket Association Stadium   
2              10         CA Lynn     Saurashtra Cricket Association Stadium   
3               6      GJ Maxwell                     Holkar Cricket Stadium   
4               0       KM Jadhav                      M Chinnaswamy Stadium   
5               9     Rashid Khan  Rajiv Gandhi International Stadium, Uppal   
6               4          N Rana                           Wankhede Stadium   
7               8        AR Patel                     Holkar Cricket Stadium   
8               0       SV Samson    Maharashtra Cricket Association Stadium   
9               4       JJ Bumrah                           Wankhede Stadium   

          umpire1        umpire2 umpire3  
0     AY Dandekar       NJ Llong     NaN  
1  A Nand Kishore         S Ravi     NaN  
2     Nitin Menon      CK Nandan     NaN  
3    AK Chaudhary  C Shamshuddin     NaN  
4             NaN            NaN     NaN  
5      A Deshmukh       NJ Llong     NaN  
6     Nitin Menon      CK Nandan     NaN  
7    AK Chaudhary  C Shamshuddin     NaN  
8     AY Dandekar         S Ravi     NaN  
9     Nitin Menon      CK Nandan     NaN

In [82]:
df_match.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 696 entries, 0 to 695
Data columns (total 18 columns):
id                 696 non-null int64
season             696 non-null int64
city               689 non-null object
date               696 non-null object
team1              696 non-null object
team2              696 non-null object
toss_winner        696 non-null object
toss_decision      696 non-null object
result             696 non-null object
dl_applied         696 non-null int64
winner             693 non-null object
win_by_runs        696 non-null int64
win_by_wickets     696 non-null int64
player_of_match    693 non-null object
venue              696 non-null object
umpire1            695 non-null object
umpire2            695 non-null object
umpire3            60 non-null object
dtypes: int64(5), object(13)
memory usage: 98.0+ KB


Info:
           1.city,winner,player_of_match,umpire1,umpire2,umpire3 columns have missing data
           2.shape - 696 rows and 18 columns

In [83]:
df_match.describe(include='all')

id       season    city        date           team1  \
count    696.000000   696.000000     689         696             696   
unique          NaN          NaN      32         498              14   
top             NaN          NaN  Mumbai  2009-05-17  Mumbai Indians   
freq            NaN          NaN      94           2              91   
mean     974.103448  2012.965517     NaN         NaN             NaN   
std     2143.239623     3.069266     NaN         NaN             NaN   
min        1.000000  2008.000000     NaN         NaN             NaN   
25%      174.750000  2010.000000     NaN         NaN             NaN   
50%      348.500000  2013.000000     NaN         NaN             NaN   
75%      522.250000  2016.000000     NaN         NaN             NaN   
max     7953.000000  2018.000000     NaN         NaN             NaN   

                              team2     toss_winner toss_decision  result  \
count                           696             696           696     696   
unique                           14              14             2       3   
top     Royal Challengers Bangalore  Mumbai Indians         field  normal   
freq                             90              90           413     686   
mean                            NaN             NaN           NaN     NaN   
std                             NaN             NaN           NaN     NaN   
min                             NaN             NaN           NaN     NaN   
25%                             NaN             NaN           NaN     NaN   
50%                             NaN             NaN           NaN     NaN   
75%                             NaN             NaN           NaN     NaN   
max                             NaN             NaN           NaN     NaN   

        dl_applied          winner  win_by_runs  win_by_wickets  \
count   696.000000             693   696.000000      696.000000   
unique         NaN              14          NaN             NaN   
top            NaN  Mumbai Indians          NaN             NaN   
freq           NaN              98          NaN             NaN   
mean      0.027299             NaN    13.472701        3.349138   
std       0.163070             NaN    23.607994        3.411398   
min       0.000000             NaN     0.000000        0.000000   
25%       0.000000             NaN     0.000000        0.000000   
50%       0.000000             NaN     0.000000        3.000000   
75%       0.000000             NaN    19.000000        6.000000   
max       1.000000             NaN   146.000000       10.000000   

       player_of_match                  venue          umpire1 umpire2  \
count              693                    696              695     695   
unique             214                     35               55      58   
top           CH Gayle  M Chinnaswamy Stadium  HDPK Dharmasena  S Ravi   
freq                20                     73               73      56   
mean               NaN                    NaN              NaN     NaN   
std                NaN                    NaN              NaN     NaN   
min                NaN                    NaN              NaN     NaN   
25%                NaN                    NaN              NaN     NaN   
50%                NaN                    NaN              NaN     NaN   
75%                NaN                    NaN              NaN     NaN   
max                NaN                    NaN              NaN     NaN   

              umpire3  
count              60  
unique             18  
top     C Shamshuddin  
freq                6  
mean              NaN  
std               NaN  
min               NaN  
25%               NaN  
50%               NaN  
75%               NaN  
max               NaN

Info:
    1. M Chinnaswamy Stadiu in Mumbai hosted most of the matches
    2. Mumbai Indians Team won ( 90 times ) toss more than other teams
    3. CH Gayle won more player of the matches than other players
    4. HDPK Dharmasena stood as a umpire for 73 matches
    

In [84]:
df_match.season.unique()

array([2017, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2018],
      dtype=int64)

Info:
    IPL Season from 2008-2018

In [85]:
df_match.city.value_counts().head()

Mumbai       94
Kolkata      70
Delhi        67
Bangalore    66
Hyderabad    56
Name: city, dtype: int64

Info:
    1. Most of the matches hosted in Mumbai city
    2. IPL Seasons Scheduled in Indian Cities and outside citiesd of India

In [86]:
df_match.result.value_counts()

normal       686
tie            7
no result      3
Name: result, dtype: int64

In [87]:
df_match.dl_applied.value_counts()

0    677
1     19
Name: dl_applied, dtype: int64

In [88]:
df_match.columns

Index(['id', 'season', 'city', 'date', 'team1', 'team2', 'toss_winner',
       'toss_decision', 'result', 'dl_applied', 'winner', 'win_by_runs',
       'win_by_wickets', 'player_of_match', 'venue', 'umpire1', 'umpire2',
       'umpire3'],
      dtype='object')

In [89]:
df_match.winner.value_counts().head()

Mumbai Indians                 98
Chennai Super Kings            90
Kolkata Knight Riders          86
Royal Challengers Bangalore    79
Kings XI Punjab                76
Name: winner, dtype: int64

In [90]:
df_match.win_by_runs.head()

0    35
1     0
2     0
3     0
4    15
Name: win_by_runs, dtype: int64

In [91]:
df_match.win_by_wickets

0       0
1       7
2      10
3       6
4       0
       ..
691     5
692     2
693     0
694     0
695     8
Name: win_by_wickets, Length: 696, dtype: int64

In [92]:
df_match.player_of_match.value_counts().head()

CH Gayle          20
AB de Villiers    18
RG Sharma         16
YK Pathan         16
DA Warner         15
Name: player_of_match, dtype: int64

In [93]:
df_match.venue.value_counts().head()

M Chinnaswamy Stadium                        73
Eden Gardens                                 70
Feroz Shah Kotla                             67
Wankhede Stadium                             66
Rajiv Gandhi International Stadium, Uppal    56
Name: venue, dtype: int64

In [94]:
df_match.umpire1.value_counts()

HDPK Dharmasena          73
Asad Rauf                51
AK Chaudhary             43
S Ravi                   40
Aleem Dar                38
BF Bowden                37
BR Doctrove              34
M Erasmus                32
RE Koertzen              20
S Asnani                 19
Nitin Menon              17
JD Cloete                16
CB Gaffaney              14
C Shamshuddin            14
AY Dandekar              13
BG Jerling               13
NJ Llong                 12
KN Ananthapadmanabhan    11
A Nand Kishore           11
SS Hazare                11
VA Kulkarni              11
RK Illingworth           11
K Hariharan              10
MR Benson                10
S Das                    10
DJ Harper                 9
GAV Baxter                7
RM Deshpande              7
IL Howell                 7
Marais Erasmus            7
BNJ Oxenford              7
Kumar Dharmasena          6
SJ Davis                  6
Bruce Oxenford            5
SK Tarapore               5
Anil Chaudhary      

In [95]:
df_match.umpire2.value_counts()

S Ravi                   56
SJA Taufel               54
C Shamshuddin            51
RJ Tucker                38
CK Nandan                36
RB Tiffin                30
VA Kulkarni              28
SK Tarapore              26
BNJ Oxenford             25
AM Saheba                23
M Erasmus                22
RE Koertzen              21
DJ Harper                18
AK Chaudhary             15
Nitin Menon              14
HDPK Dharmasena          14
K Srinath                13
NJ Llong                 13
SL Shastri               13
S Asnani                 11
K Hariharan              11
CB Gaffaney              10
VK Sharma                 9
BR Doctrove               8
PR Reiffel                8
RK Illingworth            8
IL Howell                 7
PG Pathak                 6
SS Hazare                 6
A.D Deshmukh              6
I Shivram                 6
GA Pratapkumar            6
Virender Kumar Sharma     6
Anil Chaudhary            6
AL Hill                   6
S Das               

In [96]:
df_match.umpire3.value_counts()

C Shamshuddin            6
O Nandan                 5
Nitin Menon              5
S Ravi                   5
Anil Chaudhary           5
Anil Dandekar            4
Vineet Kulkarni          4
Chris Gaffaney           3
Rod Tucker               3
A Nanda Kishore          3
Yeshwant Barde           3
A.D Deshmukh             2
Bruce Oxenford           2
Nigel Llong              2
Virender Kumar Sharma    2
Kumar Dharmasena         2
K Ananthapadmanabhan     2
Marais Erasmus           2
Name: umpire3, dtype: int64

In [97]:
df_match.columns

Index(['id', 'season', 'city', 'date', 'team1', 'team2', 'toss_winner',
       'toss_decision', 'result', 'dl_applied', 'winner', 'win_by_runs',
       'win_by_wickets', 'player_of_match', 'venue', 'umpire1', 'umpire2',
       'umpire3'],
      dtype='object')

In [98]:
#- PreProfiling
df_match.drop(['id'],axis=1,inplace=True)

In [99]:
#-Filling out missing values
df_match[df_match.city.isnull()]

season city        date                        team1  \
461    2014  NaN  2014-04-19               Mumbai Indians   
462    2014  NaN  2014-04-19        Kolkata Knight Riders   
466    2014  NaN  2014-04-23          Chennai Super Kings   
468    2014  NaN  2014-04-25          Sunrisers Hyderabad   
469    2014  NaN  2014-04-25               Mumbai Indians   
474    2014  NaN  2014-04-28  Royal Challengers Bangalore   
476    2014  NaN  2014-04-30          Sunrisers Hyderabad   

                           team2                  toss_winner toss_decision  \
461  Royal Challengers Bangalore  Royal Challengers Bangalore         field   
462             Delhi Daredevils        Kolkata Knight Riders           bat   
466             Rajasthan Royals             Rajasthan Royals         field   
468             Delhi Daredevils          Sunrisers Hyderabad           bat   
469          Chennai Super Kings               Mumbai Indians           bat   
474              Kings XI Punjab              Kings XI Punjab         field   
476               Mumbai Indians               Mumbai Indians         field   

     result  dl_applied                       winner  win_by_runs  \
461  normal           0  Royal Challengers Bangalore            0   
462  normal           0             Delhi Daredevils            0   
466  normal           0          Chennai Super Kings            7   
468  normal           0          Sunrisers Hyderabad            4   
469  normal           0          Chennai Super Kings            0   
474  normal           0              Kings XI Punjab            0   
476  normal           0          Sunrisers Hyderabad           15   

     win_by_wickets player_of_match                                venue  \
461               7        PA Patel  Dubai International Cricket Stadium   
462               4       JP Duminy  Dubai International Cricket Stadium   
466               0       RA Jadeja  Dubai International Cricket Stadium   
468               0        AJ Finch  Dubai International Cricket Stadium   
469               7       MM Sharma  Dubai International Cricket Stadium   
474               5  Sandeep Sharma  Dubai International Cricket Stadium   
476               0         B Kumar  Dubai International Cricket Stadium   

             umpire1         umpire2 umpire3  
461        Aleem Dar    AK Chaudhary     NaN  
462        Aleem Dar     VA Kulkarni     NaN  
466  HDPK Dharmasena  RK Illingworth     NaN  
468        M Erasmus          S Ravi     NaN  
469        BF Bowden       M Erasmus     NaN  
474        BF Bowden          S Ravi     NaN  
476  HDPK Dharmasena       M Erasmus     NaN

In [100]:
df_match.city.fillna("Dubai",inplace=True)

In [101]:
df_match.city.value_counts()

Mumbai            94
Kolkata           70
Delhi             67
Bangalore         66
Hyderabad         56
Chennai           49
Chandigarh        46
Jaipur            40
Pune              38
Durban            15
Centurion         12
Ahmedabad         12
Visakhapatnam     11
Rajkot            10
Indore             9
Dharamsala         9
Johannesburg       8
Dubai              7
Ranchi             7
Cape Town          7
Port Elizabeth     7
Cuttack            7
Abu Dhabi          7
Bengaluru          7
Sharjah            6
Raipur             6
Kochi              5
Kanpur             4
Mohali             3
East London        3
Nagpur             3
Kimberley          3
Bloemfontein       2
Name: city, dtype: int64

In [102]:
df_match[df_match.player_of_match.isnull()]

season       city        date                        team1  \
300    2011      Delhi  2011-05-21             Delhi Daredevils   
545    2015  Bangalore  2015-04-29  Royal Challengers Bangalore   
570    2015  Bangalore  2015-05-17             Delhi Daredevils   

                           team2                  toss_winner toss_decision  \
300                Pune Warriors             Delhi Daredevils           bat   
545             Rajasthan Royals             Rajasthan Royals         field   
570  Royal Challengers Bangalore  Royal Challengers Bangalore         field   

        result  dl_applied winner  win_by_runs  win_by_wickets  \
300  no result           0    NaN            0               0   
545  no result           0    NaN            0               0   
570  no result           0    NaN            0               0   

    player_of_match                  venue          umpire1       umpire2  \
300             NaN       Feroz Shah Kotla        SS Hazare     RJ Tucker   
545             NaN  M Chinnaswamy Stadium        JD Cloete     PG Pathak   
570             NaN  M Chinnaswamy Stadium  HDPK Dharmasena  K Srinivasan   

    umpire3  
300     NaN  
545     NaN  
570     NaN

In [103]:
#- drop no result matches
df_match[df_match.winner.isnull()]

season       city        date                        team1  \
300    2011      Delhi  2011-05-21             Delhi Daredevils   
545    2015  Bangalore  2015-04-29  Royal Challengers Bangalore   
570    2015  Bangalore  2015-05-17             Delhi Daredevils   

                           team2                  toss_winner toss_decision  \
300                Pune Warriors             Delhi Daredevils           bat   
545             Rajasthan Royals             Rajasthan Royals         field   
570  Royal Challengers Bangalore  Royal Challengers Bangalore         field   

        result  dl_applied winner  win_by_runs  win_by_wickets  \
300  no result           0    NaN            0               0   
545  no result           0    NaN            0               0   
570  no result           0    NaN            0               0   

    player_of_match                  venue          umpire1       umpire2  \
300             NaN       Feroz Shah Kotla        SS Hazare     RJ Tucker   
545             NaN  M Chinnaswamy Stadium        JD Cloete     PG Pathak   
570             NaN  M Chinnaswamy Stadium  HDPK Dharmasena  K Srinivasan   

    umpire3  
300     NaN  
545     NaN  
570     NaN

In [104]:
# Get names of indexes for which column player of the match is null;
indexNames = df_match[ df_match.player_of_match.isnull() ].index
 
# Delete these row indexes from dataFrame
df_match.drop(indexNames , inplace=True)

In [105]:
#- umpire3 - 90% of the data is missing - driop the coloumns
df_match.drop(['umpire3'],axis=1,inplace=True)

In [106]:
df_match[df_match.umpire1.isnull()]

season       city        date                        team1  \
4    2017  Bangalore  2017-04-08  Royal Challengers Bangalore   

              team2                  toss_winner toss_decision  result  \
4  Delhi Daredevils  Royal Challengers Bangalore           bat  normal   

   dl_applied                       winner  win_by_runs  win_by_wickets  \
4           0  Royal Challengers Bangalore           15               0   

  player_of_match                  venue umpire1 umpire2  
4       KM Jadhav  M Chinnaswamy Stadium     NaN     NaN

In [133]:
df_copy = df_match.copy()

In [138]:
df_copy[( df_copy.season == 2017) & (df_copy.city == "Bangalore") & (df_copy.venue == "M Chinnaswamy Stadium")]

season       city        date                        team1  \
4     2017  Bangalore  2017-04-08  Royal Challengers Bangalore   
11    2017  Bangalore  2017-04-14  Royal Challengers Bangalore   
16    2017  Bangalore  2017-04-16       Rising Pune Supergiant   
29    2017  Bangalore  2017-04-27  Royal Challengers Bangalore   
41    2017  Bangalore  2017-05-05              Kings XI Punjab   
44    2017  Bangalore  2017-05-07  Royal Challengers Bangalore   
56    2017  Bangalore  2017-05-17          Sunrisers Hyderabad   
57    2017  Bangalore  2017-05-19        Kolkata Knight Riders   

                          team2                  toss_winner toss_decision  \
4              Delhi Daredevils  Royal Challengers Bangalore           bat   
11               Mumbai Indians               Mumbai Indians         field   
16  Royal Challengers Bangalore  Royal Challengers Bangalore         field   
29                Gujarat Lions                Gujarat Lions         field   
41  Royal Challengers Bangalore  Royal Challengers Bangalore         field   
44        Kolkata Knight Riders        Kolkata Knight Riders         field   
56        Kolkata Knight Riders        Kolkata Knight Riders         field   
57               Mumbai Indians               Mumbai Indians         field   

    result  dl_applied                       winner  win_by_runs  \
4   normal           0  Royal Challengers Bangalore           15   
11  normal           0               Mumbai Indians            0   
16  normal           0       Rising Pune Supergiant           27   
29  normal           0                Gujarat Lions            0   
41  normal           0              Kings XI Punjab           19   
44  normal           0        Kolkata Knight Riders            0   
56  normal           1        Kolkata Knight Riders            0   
57  normal           0               Mumbai Indians            0   

    win_by_wickets  player_of_match                  venue  \
4                0        KM Jadhav  M Chinnaswamy Stadium   
11               4       KA Pollard  M Chinnaswamy Stadium   
16               0        BA Stokes  M Chinnaswamy Stadium   
29               7           AJ Tye  M Chinnaswamy Stadium   
41               0   Sandeep Sharma  M Chinnaswamy Stadium   
44               6        SP Narine  M Chinnaswamy Stadium   
56               7  NM Coulter-Nile  M Chinnaswamy Stadium   
57               6        KV Sharma  M Chinnaswamy Stadium   

                  umpire1        umpire2  
4                     NaN            NaN  
11  KN Ananthapadmanabhan   AK Chaudhary  
16  KN Ananthapadmanabhan  C Shamshuddin  
29           AK Chaudhary  C Shamshuddin  
41            CB Gaffaney  C Shamshuddin  
44            AY Dandekar  C Shamshuddin  
56           AK Chaudhary    Nitin Menon  
57               NJ Llong    Nitin Menon

In [131]:
def display(df_2):
    print(df_2.groupby(df_2.columns.tolist(),as_index=False).size().reset_index())
   
df_copy.groupby(['venue','city','season'])[['season','city','venue','umpire1','umpire2']].apply(display)

   season     city             venue    umpire1    umpire2  0
0    2010  Cuttack  Barabati Stadium  BF Bowden  M Erasmus  2
   season     city             venue    umpire1       umpire2  0
0    2012  Cuttack  Barabati Stadium  Aleem Dar  AK Chaudhary  1
1    2012  Cuttack  Barabati Stadium  BF Bowden   SK Tarapore  1
   season     city             venue          umpire1    umpire2  0
0    2014  Cuttack  Barabati Stadium     AK Chaudhary   NJ Llong  1
1    2014  Cuttack  Barabati Stadium  HDPK Dharmasena  PG Pathak  1
2    2014  Cuttack  Barabati Stadium         NJ Llong  CK Nandan  1
   season    city              venue          umpire1      umpire2  0
0    2010  Mumbai  Brabourne Stadium        BF Bowden    AM Saheba  1
1    2010  Mumbai  Brabourne Stadium      BR Doctrove       S Ravi  1
2    2010  Mumbai  Brabourne Stadium      BR Doctrove  SK Tarapore  1
3    2010  Mumbai  Brabourne Stadium  HDPK Dharmasena    SS Hazare  1
4    2010  Mumbai  Brabourne Stadium      RE Koertzen    RB

4    2010  Delhi  Feroz Shah Kotla        SS Hazare   SJA Taufel  1
   season   city             venue     umpire1      umpire2  0
0    2011  Delhi  Feroz Shah Kotla   AM Saheba    RB Tiffin  1
1    2011  Delhi  Feroz Shah Kotla   Asad Rauf   SL Shastri  1
2    2011  Delhi  Feroz Shah Kotla  PR Reiffel    RJ Tucker  2
3    2011  Delhi  Feroz Shah Kotla    S Asnani  RE Koertzen  1
4    2011  Delhi  Feroz Shah Kotla    S Asnani    RJ Tucker  1
   season   city             venue          umpire1        umpire2  0
0    2012  Delhi  Feroz Shah Kotla        Aleem Dar   BNJ Oxenford  1
1    2012  Delhi  Feroz Shah Kotla        Asad Rauf          S Das  1
2    2012  Delhi  Feroz Shah Kotla        Asad Rauf    SK Tarapore  1
3    2012  Delhi  Feroz Shah Kotla        BF Bowden    SK Tarapore  1
4    2012  Delhi  Feroz Shah Kotla  HDPK Dharmasena   BNJ Oxenford  1
5    2012  Delhi  Feroz Shah Kotla  HDPK Dharmasena  C Shamshuddin  1
6    2012  Delhi  Feroz Shah Kotla        JD Cloete         S Ra

   season       city                  venue          umpire1          umpire2  \
0    2012  Bangalore  M Chinnaswamy Stadium        BF Bowden    C Shamshuddin   
1    2012  Bangalore  M Chinnaswamy Stadium        BF Bowden  HDPK Dharmasena   
2    2012  Bangalore  M Chinnaswamy Stadium  HDPK Dharmasena     BNJ Oxenford   
3    2012  Bangalore  M Chinnaswamy Stadium        JD Cloete        RJ Tucker   
4    2012  Bangalore  M Chinnaswamy Stadium         S Asnani            S Das   
5    2012  Bangalore  M Chinnaswamy Stadium         S Asnani           S Ravi   
6    2012  Bangalore  M Chinnaswamy Stadium            S Das      BR Doctrove   
7    2012  Bangalore  M Chinnaswamy Stadium           S Ravi        RJ Tucker   

   0  
0  1  
1  1  
2  1  
3  1  
4  1  
5  1  
6  1  
7  1  
   season       city                  venue          umpire1        umpire2  0
0    2013  Bangalore  M Chinnaswamy Stadium        Aleem Dar  C Shamshuddin  2
1    2013  Bangalore  M Chinnaswamy Stadium      

5    VA Kulkarni  1  
   season     city                            venue         umpire1  \
0    2018  Chennai  MA Chidambaram Stadium, Chepauk  Anil Chaudhary   

          umpire2  0  
0  Chris Gaffaney  1  
   season  city                                    venue       umpire1  \
0    2015  Pune  Maharashtra Cricket Association Stadium  AK Chaudhary   
1    2015  Pune  Maharashtra Cricket Association Stadium   CB Gaffaney   
2    2015  Pune  Maharashtra Cricket Association Stadium        SD Fry   
3    2015  Pune  Maharashtra Cricket Association Stadium        SD Fry   

         umpire2  0  
0  C Shamshuddin  1  
1      K Srinath  1  
2    CB Gaffaney  1  
3      CK Nandan  1  
   season  city                                    venue      umpire1  \
0    2016  Pune  Maharashtra Cricket Association Stadium  AY Dandekar   
1    2016  Pune  Maharashtra Cricket Association Stadium  CB Gaffaney   
2    2016  Pune  Maharashtra Cricket Association Stadium  CB Gaffaney   
3    2016  Pune 

2        K Srinath  1  
   season       city                                      venue  \
0    2016  Hyderabad  Rajiv Gandhi International Stadium, Uppal   
1    2016  Hyderabad  Rajiv Gandhi International Stadium, Uppal   
2    2016  Hyderabad  Rajiv Gandhi International Stadium, Uppal   
3    2016  Hyderabad  Rajiv Gandhi International Stadium, Uppal   
4    2016  Hyderabad  Rajiv Gandhi International Stadium, Uppal   
5    2016  Hyderabad  Rajiv Gandhi International Stadium, Uppal   

           umpire1          umpire2  0  
0     AK Chaudhary        CK Nandan  2  
1     AK Chaudhary  HDPK Dharmasena  1  
2      AY Dandekar        CK Nandan  1  
3  HDPK Dharmasena        VK Sharma  1  
4       K Bharatan        M Erasmus  1  
5        M Erasmus           S Ravi  1  
   season       city                                      venue  \
0    2017  Hyderabad  Rajiv Gandhi International Stadium, Uppal   
1    2017  Hyderabad  Rajiv Gandhi International Stadium, Uppal   
2    2017  Hyderab

7    2012  Mumbai  Wankhede Stadium        S Ravi     RJ Tucker  1
   season    city             venue          umpire1       umpire2  0
0    2013  Mumbai  Wankhede Stadium     AK Chaudhary    SJA Taufel  1
1    2013  Mumbai  Wankhede Stadium        Asad Rauf  AK Chaudhary  1
2    2013  Mumbai  Wankhede Stadium        Asad Rauf      S Asnani  2
3    2013  Mumbai  Wankhede Stadium  HDPK Dharmasena     CK Nandan  1
4    2013  Mumbai  Wankhede Stadium  HDPK Dharmasena        S Ravi  1
5    2013  Mumbai  Wankhede Stadium        M Erasmus   VA Kulkarni  1
6    2013  Mumbai  Wankhede Stadium           S Ravi    SJA Taufel  1
   season    city             venue          umpire1        umpire2  0
0    2014  Mumbai  Wankhede Stadium     BNJ Oxenford  C Shamshuddin  1
1    2014  Mumbai  Wankhede Stadium  HDPK Dharmasena      RJ Tucker  1
2    2014  Mumbai  Wankhede Stadium  HDPK Dharmasena    VA Kulkarni  1
3    2014  Mumbai  Wankhede Stadium        K Srinath      RJ Tucker  1
4    2014  Mumbai 

Empty DataFrame
Columns: []
Index: []